internal/error.go

```diff
@@ -303,6 +322,28 @@ func NewApplicationError(msg string, errType string, nonRetryable bool, cause er
 	return applicationErr
 }
 
+func NewApplicationErrorWithExtraInformation(errorDescriptors ErrorAttributes, extraInformation ExtraInformation) error {
```

Comment:
To match other parts of the Go SDK, can this be:

```go
type ApplicationErrorOptions struct {
    Details []any
    Cause error
    NonRetryable bool
    NextAttemptRetryDelay time.Duration
}

func NewApplicationErrorWithOptions(message, errType string, options ApplicationErrorOptions) error
```

Reply:
I would also return `*ApplicationError` here instead of the `error` interface.